ROADMAP.md

```diff
@@ -0,0 +1,118 @@
+
+
+# go-libp2p roadmap Q4’22/Q1’23
+
+```
+Date: 2022-10-07
+Status: In Progress
+Notes: This document is still in review will may be heavily modified based on stakeholder feedback. Please add any feedback or questions in:
+https://github.com/libp2p/go-libp2p/issues/1806
+```
+
+## 🛣️ Milestones
+### 2022
+
+#### Early Q4 (October)
+-   A.1⚡Handshakes at the Speed of Light - **[Early Muxer Negotiation](https://github.com/libp2p/specs/issues/426)**
+- E.1 🧪 Future-proof testing - **RTT test for Early Muxer Negotiation**
+
+### Mid Q4 (November)
+-   F.1 ⏱ Future-proof Benchmarking - **[Using nix-builders](https://github.com/testground/testground/pull/1425)**
+
+#### End of Q4 (December)
+-   B.1 🧠 Smart Dialing - **Happy Eyeballs**
+-   C 📊 Comprehensive Metrics - **Swarm metrics**
+-   D.1 📺 Universal Browser Connectivity - **[WebRTC Browser -> Server](https://github.com/libp2p/go-libp2p/pull/1655)**
+
+### 2023
+
+#### Early Q1 (January)
+-   B.2 🧠 Smart Dialing - **QUIC Blackhole detector**
+-   A.2 ⚡ Handshakes at the Speed of Light - **[Adding security protocol](https://github.com/libp2p/specs/pull/353)**
+
+#### Mid Q1 (February)
+-   B.3 🧠 Smart Dialing - **RTT estimation**
+    -   🎉 Estimated Project Completion
+
+#### End of Q1 (March)
+-   F.2 ⏱️ Future-proof benchmarking - **Using first class support for remote runners in Testground**
+-   A.3 ⚡ Handshakes at the Speed of Light - **0.5 RTT data optimization** (for QUIC)
+    -   🎉 Estimated Project Completion
+
+#### Up Next
+-   D.2 📺 Universal Browser Connectivity - **[WebTransport](https://github.com/libp2p/go-libp2p/issues/1717)** update to new draft versions
+    - Dependency on Chrome needs to support new draft version of WebTransport protocol (still under development by IETF and W3C)
+-  E.2 🧪 Future-proof testing - **Comprehensive testing for non-transport features**
+-  G 📢 Judicious Address Advertisements
+
+# Roadmap Appendix
+## [A.⚡️Handshakes at the Speed of Light](https://github.com/libp2p/go-libp2p/issues/1807)
+
+**Why**: Historically, libp2p has been very wasteful when it comes to round trips spent during connection establishment. This is slowing down our users, especially their TTFB (time to first byte) metrics.
```

Comment:
I agree this is true.  I would love to have more insight on the actual impact.  For example, it would be great to have anecdotes from users on how much this is an adoption blocker.

This is more general comment.  I don't think there's necessarily anything to do right now.  It's more just to push to find ways to quantify problems in terms of impact as much as possible.

I do agree this issue rolls into one of our key pillars of "performance".  TTFB performance isn't the existential threat I have heard complaints about though (vs. libp2p is slower than HTTP).  I could be tuned in here though.  Getting user input would be great.

Reply:
I believe TTFB is an important metric to both Saturn and ipfs cluster folks.